src/Linux/Host/Broker/Impl/Startup/LinuxStartup.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.Common.Core.IO;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.R.Host.Broker.Services;
+using Microsoft.R.Interpreters;
+using Microsoft.R.Common.Core.Linux;
+
+namespace Microsoft.R.Host.Broker.Startup
+{
+    public sealed class LinuxStartup : Startup {
+        public LinuxStartup(ILoggerFactory loggerFactory, IConfigurationRoot configuration) : base(loggerFactory, configuration) { }
+
+        public override void ConfigureServices(IServiceCollection services) {
+            base.ConfigureServices(services);
+
+            var fs = new FileSystem();
+            services.AddSingleton<IFileSystem>(fs)
+                .AddSingleton<IAuthenticationService, LinuxAuthenticationService>()
```

Comment:
If it is expected to be frequently used service, you can create and add actual instance and avoid reflection-based creation. I.e. .AddService(new Foo())